lib/rust/ensogl/core/src/display/object/instance.rs

```diff
@@ -1490,13 +1491,23 @@ pub struct HierarchyFrp {
     /// removed from a visible parent or added to an invisible one.
     pub on_hide:            frp::Stream<Option<Scene>>,
     /// Fires during the first scene refresh if this object was moved between scene layers.
-    pub on_layer_change:    frp::Stream<(Option<Scene>, Option<WeakLayer>, Option<WeakLayer>)>,
+    pub on_layer_change: frp::Stream<(
+        Option<Scene>,
+        Option<WeakLayer>,
+        Option<WeakLayer>,
+        Option<VirtualSublayerAssignment>,
```

Comment:
I have here a small problem with inconsistent naming - we have `WeakLayer` and `VirtualSublayerAssignment`, which use different naming styles. We can rename `WeakLayer` to `LayerAssignment` or something similar. I'm not sure though if my proposal is correct. This is just a first impression after seeing this part of the code, but I might be totally wrong, so feel free to skip this comment.